test/evergreen.yml

```diff
@@ -755,6 +746,42 @@ functions:
 #######################################
 
 variables:
+  # Configure flags for builtins.
+  - &configure_flags_with_builtins
+    HAVE_BUILTIN_EXTENSION_LZ4: -DHAVE_BUILTIN_EXTENSION_LZ4=1
+    HAVE_BUILTIN_EXTENSION_SNAPPY: -DHAVE_BUILTIN_EXTENSION_SNAPPY=1
+    HAVE_BUILTIN_EXTENSION_ZLIB: -DHAVE_BUILTIN_EXTENSION_ZLIB=1
+    HAVE_BUILTIN_EXTENSION_ZSTD: -DHAVE_BUILTIN_EXTENSION_ZSTD=1
+
+  # Configure flags static library (default in cmake is dynamic).
+  - &configure_flags_static_lib
+    ENABLE_SHARED: -DENABLE_SHARED=0
+    ENABLE_STATIC: -DENABLE_STATIC=1
+
+  # Configure flags static library (default in cmake is dynamic) with builtins.
+  - &configure_flags_static_lib_with_builtins
```

Comment:
I like putting the configure flags in variables, but I'm not sure about the "variables that include other variables" thing. Stuff like builtins and static are orthogonal, and I think it'd be clearer to have tasks use different variables for different bits of config. Stuff like `configure_flags_address_sanitizer_mongodb_v4_clang` below are IMO a bit much (the name also doesn't include the fact that it uses builtins, which would make it even more difficult to read).

Reply:
I can update the name to make it clearer. But with evergreen, you can't have two uses of vars (<<: syntax) in one section so you can't use variables cleanly in a way. Its actually possible for us to think about making the sanitizer a build variant and make that much cleaner (by putting the build type in the variant definition) which would stop some of this duplication. I can look at that.

Reply:
Thanks - that sounds like a better way to do it. That's a shame about not having multiple variables though.

Reply:
There isn't a clean path because the tests that use the sanitisers aren't always in a sanitiser variant. Updated the naming to at least be consistent.

Reply:
Thanks. It's better now, shame about the lack of multiple variables but nothing we can do about it in this PR.